cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/validator/AwsGatewaySubnetMultiAzValidator.java

```diff
@@ -0,0 +1,110 @@
+import com.amazonaws.services.ec2.model.DescribeSubnetsRequest;
+import com.amazonaws.services.ec2.model.DescribeSubnetsResult;
+import com.amazonaws.services.ec2.model.Subnet;
+import com.sequenceiq.cloudbreak.cloud.Validator;
+import com.sequenceiq.cloudbreak.cloud.aws.common.CommonAwsClient;
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonEc2Client;
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AwsCredentialView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
+import com.sequenceiq.cloudbreak.cloud.model.Group;
+import com.sequenceiq.cloudbreak.cloud.model.GroupNetwork;
+import com.sequenceiq.cloudbreak.cloud.model.GroupSubnet;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+
+@Component
+public class AwsGatewaySubnetMultiAzValidator implements Validator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsGatewaySubnetMultiAzValidator.class);
+
+    @Inject
+    private CommonAwsClient awsClient;
+
+    @Override
+    public void validate(AuthenticatedContext ac, CloudStack cloudStack) {
+        LOGGER.debug("Validating that only one subnet per availability zone have been provided for the gateway group.");
+        AwsCredentialView awsCredential = new AwsCredentialView(ac.getCloudCredential());
+        String region = ac.getCloudContext().getLocation().getRegion().value();
+        Optional<Group> gatewayGroupOptional = getGatewayGroup(cloudStack);
+        if (gatewayGroupOptional.isPresent()) {
+            Group group = gatewayGroupOptional.get();
+            Set<String> subnetIds = getInstanceGroupNetworSubnetIds(group);
+            if (!subnetIds.isEmpty()) {
+                Map<String, Set<String>> subnetIdSetsByAvailabilityZone = fetchingSubnetsAndGroupingByAvailabilityZone(awsCredential, subnetIds, region);
+                Set<String> availabilityZonesWithMultipleSubnets = subnetIdSetsByAvailabilityZone.entrySet()
+                        .stream()
+                        .filter(entry -> entry.getValue().size() > 1)
+                        .map(e -> String.format("subnets: '%s' from availability zone: '%s'", String.join(",", e.getValue()), e.getKey()))
+                        .collect(toSet());
+
+                if (!availabilityZonesWithMultipleSubnets.isEmpty()) {
+                    LOGGER.debug("Validation error found there are multiple subnets associated with an availability zone");
+                    String validationError = String.format("Only one subnet per Availability Zone can be specified for a Gateway group. The request contains %s",
+                            String.join(", ", availabilityZonesWithMultipleSubnets));
+                    LOGGER.info(validationError);
+                    throw new CloudConnectorException(validationError);
+                }
```

Comment:
maybe a valid message would be useful

Reply:
https://github.com/hortonworks/cloudbreak/pull/11122/files#diff-8a1bfaa46ebbdd0eca8379e936a207e4c6562801defbc47d2bfaee0902eb9921R69